src/helpers.ts

```diff
@@ -142,34 +142,57 @@ export function isDoNotMergeLabel(string: string): boolean {
   return match != null
 }
 
+async function pullRequestsMatching(
+  octokit: Octokit,
+  repo: Repo,
+  branch: String | null,
+  sha: String
+): Promise<number[]> {
+  const repoOwner = repo.owner?.login
+
+  if (!repoOwner) return []
+
+  const pullRequests = (
+    await octokit.pulls.list({
+      ...github.context.repo,
+      state: 'open',
+      head: `${repoOwner}:${branch}`,
+      sort: 'updated',
+      direction: 'desc',
+      per_page: 100,
+    })
+  ).data as PullRequest[]
+
+  return pullRequests.filter(pr => pr.head.sha === sha).map(({ number }) => number)
+}
+
+export async function pullRequestsForCheckSuite(octokit: Octokit, checkSuite: CheckSuite): Promise<number[]> {
+  let pullRequests = (checkSuite.pull_requests as PullRequest[]).map(({ number }) => number)
```

Comment:
Why does this need to be cast here?

Reply:
Well, the code was mostly copypasted from `pullRequestsForWorkflowRun`, which has had this conversion.
I'm not good at TypeScript, so didn't change much.

Omitting the conversion leads to the error message `Object is possibly null`, so perhaps it makes sense to replace the cast with `|| []`.
I could do that if it's fine with you.

Reply:
I don't think the cast is necessary anymore, try:

```ts
checkSuite.pull_requests?.map(({ number }) => number) ?? []
```

If that works, please also adjust this in `pullRequestsForWorkflowRun`.